### PR TITLE
 details: implement dependency path viz 

### PR DESCRIPTION
**Pull request by Fishrock123**
This is a thing we've done elsewhere and there is no reason not
to also do it here since it is useful and universal-module-tree
exposes the data.

Closes: https://github.com/nodesource/ncm-cli/issues/74
Refs: https://github.com/nodesource/ncm-cli/issues/120

### CODE REVIEW

File: commands/details.js [hunk 1]
```diff
@@ -46,9 +50,23 @@ async function details (argv, arg1, arg2, arg3) {
     return
   }
 
+  const tree = await universalModuleTree(dir)
+  const list = universalModuleTree.flatten(tree)
```
Comment (juliangruber) [inline commands/details.js]: fyi this is also the return type of `ncm-analyze-tree`, if you want to reuse that instead.

File: commands/details.js [hunk 2]
```diff
@@ -46,9 +50,23 @@ async function details (argv, arg1, arg2, arg3) {
...
+  }
+
   /* NCM-Cli Header */
   L()
-  L(header(`${name} @ ${version}`))
+  if (requirePaths.length > 0) {
+    L(header(`${name} @ ${version} (within ${path.basename(dir)})`))
+  } else {
+    L(header(`${name} @ ${version}`))
+  }
```
Comment (juliangruber) [inline commands/details.js]: for reviewing this it would be handy if you attached screenshots of the command output in the PR

**Comment by Fishrock123** [comment]
(Within `ncm-cli`)
```
bin/ncm-cli.js details chalk @ 2.4.1
```
Screen Shot 2019-03-25 at 12 36 22 PM

File: commands/details.js [hunk 3]
```diff
@@ -25,6 +27,8 @@ async function details (argv, arg1, arg2, arg3) {
     printHelp()
     return
   }
+  let { dir } = argv
```
Comment (Raynos) [inline commands/details.js]: can we document `--dir` in the help output ?

**Pull request merged**